File: src/main/java/pl/tsimur/jpatest/model/dto/LoginDto.java
package pl.tsimur.jpatest.model.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.Email;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class LoginDto {
    @Email
    private String email;
    private String password;
}

File: src/main/kotlin/pl/timur/jpatest/model/dto/LoginDto.kt
package pl.timur.jpatest.model.dto

import jakarta.validation.constraints.Email
import pl.timur.jpatest.model.User

data class LoginDto(
    @Email
    val email: String,
    val password: String
) {
    fun toUser() = User(email = email, password = password)
}
